src/OperationalInsights/OperationalInsights/Models/DataSourcePropertiesPerKind/PSCustomLogDataSourceProperties.cs

```diff
@@ -163,7 +163,12 @@ public class DateTimeExtraction
         /// <summary>
         /// Gets or sets the regex.
         /// </summary>
-        public string regex { get; set; }
+        public RegexDelimiter[] regex { get; set; }
```

Comment:
I think thsi model will need to be populated by cmdlet code as well

Reply:
Why?
This feature is in deprecation mode in its current form, therefore we don't want to invest too much in this. Currently just fix the bug, we will discuss our PMs if this is needed.

Reply:
First, it is a breaking change, if this is returned from your cmdlet, which is not allowed.  Second, how is this value actually populated?

Reply:
1. This is not a breaking change, this is a bug fix. You can see [here the ICM related incident](https://icm.ad.msft.net/imp/v3/incidents/details/121625584/home).  
2. The following object is used in two locations:
a. **In New-AzOperationalInsightsCustomLogDataSource cmdlet:** 
This cmdlet fails because of BadRequest because the backend expects a different contract. After this fix (validated of course) the request succeeds. This is populated by supplying a raw json of the CustomLog model. Example:
```powershell
$customLogRawJson = Get-Content 'C:\Users\arolshan\Desktop\customLogRaw.json' | Out-String
New-AzOperationalInsightsCustomLogDataSource -ResourceGroupName "arolshan-eus-rg" -WorkspaceName "arolshan-eus-ws2" -Name "arolshan_datetime_CL" -CustomLogRawJson $customLogRawJson
```
where $customLogRaw.json holds:
```json
{
    "CustomLogName": "arolshan_datetime_CL",
    "Description": "this is a description",
    "Extractions": [
        {
            "extractionName": "TimeGenerated",
            "extractionProperties": {
                "dateTimeExtraction": {
                    "regex": [
                        {
                            "matchIndex": 0,
                            "numberdGroup": null,
                            "pattern": "((\\d{2})|(\\d{4}))-([0-1]\\d)-(([0-3]\\d)|(\\d))\\s((\\d)|([0-1]\\d)|(2[0-4])):[0-5][0-9]:[0-5][0-9]"
                        }
                    ]
                }
            },
            "extractionType": "DateTime"
        }
    ],
    "Inputs": [
        {
            "location": {
                "fileSystemLocations": {
                    "linuxFileTypeLogPaths": null,
                    "windowsFileTypeLogPaths": [
                        "c:\\tmp\\datetimelog.txt"
                    ]
                }
            },
            "recordDelimiter": {
                "regexDelimiter": {
                    "matchIndex": 0,
                    "numberdGroup": null,
                    "pattern": "(^.*((\\d{2})|(\\d{4}))-([0-1]\\d)-(([0-3]\\d)|(\\d))\\s((\\d)|([0-1]\\d)|(2[0-4])):[0-5][0-9]:[0-5][0-9].*$)"
                }
            }
        }
    ]
}
```


b. **In Get-AzOperationalInsightsDataSource cmdlet:**
This also fails when trying to perform the following:
```powershell
Get-AzOperationalInsightsDataSource -ResourceGroupName "arolshan-eus-rg" -WorkspaceName "arolshan-eus-ws2" -Kind "CustomLog" 
```
on deserialization, because the backend sends a different model than the one the powershell module expects.

Reply:
Removing a property or changing its type from the public API surface is, **by definition** a breaking change.  Is the contention that this never worked, or that the service changed its behavior for this api-version?

You are deciding in this cmdlet to simply serialize or deserialize using this model, that is an implementation choice - it should be possible for you to fix the bug without breaking the public API surface.

Also, if this is  a bug fix, you should add a regression test, and this will also help to demonstrate the actual usage of this model type in the cmdlet.

Reply:
> Removing a property or changing its type from the public API surface is, by definition a breaking change. Is the contention that this never worked, or that the service changed its behavior for this api-version?

Unfortunately this never worked and it took one customer to try it to notice the bug. 

> You are deciding in this cmdlet to simply serialize or deserialize using this model, that is an implementation choice - it should be possible for you to fix the bug without breaking the public API surface.

You are absolutely right, but I don't see the point of allowing backward compatibility for something that doesn't work. 

> Also, if this is a bug fix, you should add a regression test, and this will also help to demonstrate the actual usage of this model type in the cmdlet.

Your'e right, I'll modify current tests to actually test the functionality properly. 

> If something has changed in the underlying API, then you need to mitigate the break, which would mean keeping the current string property, adding a new property with the new value, and ensuring that the properties are populated by the cmdlet. Adding a CmdletBreakingChangeAttribute to the model property.

a. See point I made above. 
b. Sure I'll add CmdletBreakingChangeAttribute to the model.

Reply:
The point is that breaking changes are not allowed.  A customer can stiull write a script that uses this propety, even if it is null.  You need to add a property and not remove this one.  ALso, if it is possible to recreate this string based on the returned values (and looks to me like it might be), you should do that.

Reply:
mark, I added a json converter specifically for this case. If a user uses the script using the old property then the json converter will convert it to the right property. please see **RegexDelimiterJsonConverter**

Reply:
I added the old property back. Please see.